pagerduty/resource_pagerduty_event_rule.go

```diff
@@ -0,0 +1,145 @@
+package pagerduty
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func resourcePagerDutyEventRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePagerDutyEventRuleCreate,
+		Read:   resourcePagerDutyEventRuleRead,
+		Update: resourcePagerDutyEventRuleUpdate,
+		Delete: resourcePagerDutyEventRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"action_json": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"condition_json": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"advanced_condition_json": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"catch_all": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func buildEventRuleStruct(d *schema.ResourceData) *pagerduty.EventRule {
+	eventRule := &pagerduty.EventRule{
+		Actions:   expandString(d.Get("action_json").(string)),
+		Condition: expandString(d.Get("condition_json").(string)),
+	}
+
+	if attr, ok := d.GetOk("advanced_condition_json"); ok {
+		eventRule.AdvancedCondition = expandString(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("catch_all"); ok {
+		eventRule.CatchAll = attr.(bool)
+	}
+
+	return eventRule
+}
+
+func expandString(v string) []interface{} {
+	var obj []interface{}
+	err := json.Unmarshal([]byte(v), &obj)
+
+	if err != nil {
+		log.Printf(string(err.Error()))
+	}
+
+	return obj
+}
+
+func flattenSlice(v []interface{}) string {
+	b, err := json.Marshal(v)
+	if err != nil {
+		log.Printf(string(err.Error()))
+	}
+	return string(b)
+}
+
+func resourcePagerDutyEventRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*pagerduty.Client)
+
+	eventRule := buildEventRuleStruct(d)
+
+	log.Printf("[INFO] Creating PagerDuty event rule: %s", "eventRule")
+
+	eventRule, _, err := client.EventRules.Create(eventRule)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(eventRule.ID)
+
+	return resourcePagerDutyEventRuleRead(d, meta)
+}
+
+func resourcePagerDutyEventRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*pagerduty.Client)
+
+	log.Printf("[INFO] Reading PagerDuty event rule: %s", d.Id())
+
+	resp, _, err := client.EventRules.List()
+	if err != nil {
+		return err
+	}
+	for _, rule := range resp.EventRules {
```

Comment:
What do you think about doing something like:

```go
var found *pagerduty.EventRule

for _, rule := range resp.EventRules {
  if rule.ID == d.Id() {
    found = rule
    break
  }
}

if found == nil {
  d.SetId("")
}
...
d.Set("catch_all", rule.CatchAll)
...
```


That way you can set the attributes outside the loop 🙂

Reply:
This makes sense. I've adopted this idea in my next commit.